library/dash/src/main/java/com/google/android/exoplayer2/source/dash/DashMediaPeriod.java

```diff
@@ -369,7 +369,19 @@ private int[] getStreamIndexToTrackGroupIndex(ExoTrackSelection[] selections) {
     int[] streamIndexToTrackGroupIndex = new int[selections.length];
     for (int i = 0; i < selections.length; i++) {
       if (selections[i] != null) {
-        streamIndexToTrackGroupIndex[i] = trackGroups.indexOf(selections[i].getTrackGroup());
+        int trackIndex = trackGroups.indexOf(selections[i].getTrackGroup());
+
+        // fallback in case we could not find map trackGroups through reference
+        if(trackIndex == -1) {
+          for(int y = 0; y < trackGroups.length; y++) {
+            if(trackGroups.get(y).getFormat(0).id == selections[i].getTrackGroup().getFormat(0).id) {
+              trackIndex = y;
+              break;
+            }
+          }
+        }
+
+        streamIndexToTrackGroupIndex[i] = trackIndex;
```

Comment:
if trackIndex still equal -1 , which means the loop didnt find it by id then instead of -1 we must set it to C.INDEX_UNSET